typing/types.ml

```diff
@@ -167,7 +167,7 @@ and record_representation =
   | Record_float                        (* All fields are floats *)
   | Record_unboxed of bool    (* Unboxed single-field record, inlined or not *)
   | Record_inlined of int               (* Inlined record *)
-  | Record_extension                    (* Inlined record under extension *)
+  | Record_extension of Path.t          (* Inlined record under extension *)
```

Comment:
My guess would be: to avoid doing things [like this](https://github.com/ocaml/ocaml/pull/1610/files?diff=unified#diff-e8899086c6b3526d6bc4453b4ad5395cL1304).
That is: unless I missunderstood, it is a cleanup unrelated to the rest of this PR.